zproject/prod_settings_template.py

```diff
@@ -209,6 +210,27 @@
 #
 #SOCIAL_AUTH_SUBDOMAIN = 'auth'
 
+########
+# Cognito OAuth.
+#
+# To set up Cognito authentication, you'll need to do the following:
+#
+# (1) Visit https://console.aws.amazon.com/cognito/home , and select
+# "Manager User Pools".
+#
+# (2) Choose an existing pool or create a now one following the Cognito Pool Tutorial
+# https://docs.aws.amazon.com/cognito/latest/developerguide/tutorial-create-user-pool.html
+#
+# (3) Create an app and configure a pool domain
+#
+# (4) You should get a client ID and a client secret. Copy them.
+# Use the client ID as `SOCIAL_AUTH_COGNITO_KEY` here, and put the
+# client secret in zulip-secrets.conf as `social_auth_cognito_secret`.
+#SOCIAL_AUTH_COGNITO_KEY = <your client ID from Cognito>
+#
+# (5) Provide the Cognito domain as SOCIAL_AUTH_COGNITO_POOL_DOMAIN here.
+#SOCIAL_AUTH_COGNITO_POOL_DOMAIN = example.com
```

Comment:
I think given that this backend is a bit more obscure, we should move it after SAML in the file, and also move these docs to the ReadTheDocs site; we've generally been moving away from detailed docs here because they're harder to update.